### PR TITLE
multiple updates for RT and schedule rewrites

### PR DESCRIPTION
**Pull request by atvaccaro**
Changes necessary to address comments on https://github.com/cal-itp/data-infra/pull/1648 (reducing metadata).

Fixing request building bug(s) (specifically around jinja templating and not also removing trailing ampersands).

Configuring buckets via env vars; this is something we should be doing in general.

Clears gcsfs cache when identifying latest files such as Airtable GTFS datasets.

### CODE REVIEW

**Review by lauriemerrell — APPROVED** [review]
👍 🙏  tiniest nit that it is now `8.9` re: calendar version but 🤷

File: calitp/storage.py [hunk 1]
```diff
@@ -99,17 +99,9 @@ def make_name_bq_safe(name: str):
     return str.lower(re.sub("[^\w]", "_", name))  # noqa: W605
 
 
-def prefix_bucket(bucket):
```
Comment (lauriemerrell) [inline calitp/storage.py:102]: This is used a bunch of other places in data-infra, ex. `airtable_loader_v2` and `create_external_tables`... Will you update those too or do we want to leave this to support those use cases?

File: calitp/storage.py [hunk 2]
```diff
@@ -672,9 +565,10 @@ def download_feed(
                 cls=GTFSFeedExtractInfo,
                 fs=get_fs(),
                 partitions={
-                    "dt": pendulum.today().date(),
+                    "dt": pendulum.yesterday().date(),
```
Comment (lauriemerrell) [inline calitp/storage.py:568]: isn't this already used by RT in v3 "prod"? what effect will this change have / why change from today to yesterday?

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
a few questions

File: calitp/storage.py [hunk 3]
```diff
@@ -99,17 +99,9 @@ def make_name_bq_safe(name: str):
...
-    bucket = bucket.replace("gs://", "")
-    return f"gs://test-{bucket}" if is_development() else f"gs://{bucket}"
-
-
-AIRTABLE_BUCKET = prefix_bucket("gs://calitp-airtable")
-SCHEDULE_RAW_BUCKET = prefix_bucket("gs://calitp-gtfs-schedule-raw")
-RT_RAW_BUCKET = prefix_bucket("gs://calitp-gtfs-rt-raw")
-SCHEDULE_PROCESSED_BUCKET = prefix_bucket("gs://calitp-gtfs-schedule-processed")
+AIRTABLE_BUCKET = os.getenv("CALITP_BUCKET__AIRTABLE")
+SCHEDULE_RAW_BUCKET = os.getenv("CALITP_BUCKET__SCHEDULE_RAW")
```
Comment (lauriemerrell) [inline calitp/storage.py:103]: Ok sorry I know you ran these by me but I missed it until I went to set one of these myself -- I would call it `CALITP_BUCKET__GTFS_SCHEDULE_RAW` i.e., specify `GTFS`, and same for RT immediately below

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
one more thing

File: calitp/storage.py (hunk 1 above)
Comment (atvaccaro) [reply calitp/storage.py:102]: I am changing those over for Airtable (and RT) jobs, but the external tables use a flag that's just named the same thing

File: calitp/storage.py (hunk 2 above)
Comment (atvaccaro) [reply calitp/storage.py:579]: This is just for local testing, it's not actually used anywhere

**Review by lauriemerrell — APPROVED** [review]

**Pull request merged**